### PR TITLE
RE2022-203: Simplify & refactor load ver overrides & subset application

### PR DESCRIPTION
**Pull request by MrCreosote**
* Simplifies the code for overriding load versions slightly to make it easier to understand what's going on.
* Makes the match & selection appliction code in the heatmap data product (mostly) static in preparation for moving it to a shared module for reuse with samples.

### CODE REVIEW

File: src/service/data_products/genome_attributes.py [hunk 1]
```diff
@@ -241,7 +241,7 @@ async def get_genome_attributes(
     sort_desc: common_models.QUERY_VALIDATOR_SORT_DIRECTION = False,
     skip: common_models.QUERY_VALIDATOR_SKIP = 0,
     limit: common_models.QUERY_VALIDATOR_LIMIT = 1000,
-    output_table: common_models.QUERY_VALIDATOR_OUTPUT_TABLE = False,
+    output_table: common_models.QUERY_VALIDATOR_OUTPUT_TABLE = True,
```
Comment (MrCreosote) [inline src/service/data_products/genome_attributes.py:244]: Accidentally swapped this in the last PR

File: src/service/data_products/heatmap.py [hunk 2]
```diff
@@ -317,54 +319,45 @@ async def _get_load_version_and_processes( # pretty huge method sig here
...
-            load_ver = await get_load_version(
-                appstate.arangostorage, collection_id, data_product, load_ver_override, user)
+        lvo = override_load_version(load_ver_override, match_id, selection_id)
+        coll, load_ver = await get_load_version(
+            appstate.arangostorage, collection_id, data_product, lvo, user)
         if match_id:
             dp_match = await processing_matches.get_or_create_data_product_match_process(
-                appstate, coll, user, match_id, data_product, subset_fn
+                appstate, coll, user, match_id, data_product,
+                partial(self._process_subset, collection)
```
Comment (MrCreosote) [inline src/service/data_products/heatmap.py:339]: I'll get rid of `self` references in this method when I move it and `_process_subset` to a new module

Comment (Tianhao-Gu) [reply src/service/data_products/heatmap.py:339]: 👍

**Review by Tianhao-Gu — DISMISSED** [review]
LGTM besides a typo.

File: src/service/data_products/common_functions.py [hunk 3]
```diff
@@ -15,19 +15,32 @@
 from typing import Any, Callable
 
 
+def override_load_version(
+    load_ver_override: str = None, match_id: str = None, selection_id: str = None
+) -> str | None:
+    """
+    Determines whether a load verison override should be used for subsequent method calls. Returns
```
[the remark this answers is not in the document]
Comment (MrCreosote) [reply src/service/data_products/common_functions.py]: fixed

File: src/service/data_products/common_functions.py [hunk 4]
```diff
@@ -15,19 +15,32 @@
...
+
+
 async def get_load_version(
     store: ArangoStorage,
     collection_id: str,
     data_product: str,
     load_ver: str,
     user: kb_auth.KBaseUser,
-) -> str:
+) -> tuple[models.ActiveCollection, str]:
```
[the remark this answers is not in the document]
Comment (MrCreosote) [reply src/service/data_products/common_functions.py]: fixed

**Review by Tianhao-Gu — APPROVED** [review]
👍

**Pull request merged**